Deprecate custom ID generation APIs from Wonder Blocks

somewhatabstract opened the pull request:
## Summary:
Welcome to the Wonder Blocks 1812 Overture - a series of PRs that will deprecate and remove the custom ID generation utilities in favour of React's `useId` and the new `Id` component that is built atop of it.

This updates dependencies in Perseus to latest webapp dependencies (used the sync script for that), and includes the soon to be released Wonder Blocks that deprecates the custom ID generation utilities in favour of React's `useId` and the new `Id` component that is built atop of it.

An upcoming PR will then have the work of removing the old API uses so that we can then delete them from a future Wonder Blocks release.

NOTE: I will update this to the actual released WB before landing, once that happens. I am trying to get everything reviewed and ready to go so that when I do that release, I can get everything up-to-date quickly and correctly.

Issue: WB-1812

## Test plan:
`yarn test`
`yarn lint`

catandthemachines reviewed (approved):
Approved! Added note, @somewhatabstract the console errors are refering to the use of SingleSelect on `answer-choices.tsx`. Just update line 33 with `opener: (openerProps: OpenerProps) => React.ReactElement<any>;`.

somewhatabstract commented:
> Approved! Added note, @somewhatabstract the console errors are refering to the use of SingleSelect on `answer-choices.tsx`. Just update line 33 with `opener: (openerProps: OpenerProps) => React.ReactElement<any>;`.

I fixed this with what I think is a better approach that matches the intent of the broken declaration. I used `PropsFor` from Wonder Blocks Core to get the props type for `SingleSelect` and then indexed the type of the `opener` prop. That way, this component will always match that even if the `SingleSelect` type gets modified in some future change.
